packages/react-components/src/trace-explorer/trace-explorer-opened-traces-widget.tsx

```diff
@@ -235,15 +228,33 @@ export class ReactOpenTracesWidget extends React.Component<ReactOpenTracesWidget
 
     protected async doUpdateOpenedExperiments(): Promise<void> {
         const remoteExperiments = await this._experimentManager.getOpenedExperiments();
+
         remoteExperiments.forEach(experiment => {
             this._experimentManager.addExperiment(experiment);
         });
+
+        remoteExperiments.forEach(newExp => {
+            this.state.openedExperiments.forEach(oldExp =>{
+                if (newExp.UUID === oldExp.UUID) {
+                    newExp.name = oldExp.name;
+                }
+            });
+        });
+
         const selectedIndex = remoteExperiments.findIndex(experiment => this._selectedExperiment &&
             experiment.UUID === this._selectedExperiment.UUID);
         this.setState({ openedExperiments: remoteExperiments, selectedExperimentIndex: selectedIndex !== -1 ? selectedIndex : 0 });
         signalManager().fireOpenedTracesChangedSignal(new OpenedTracesUpdatedSignalPayload(remoteExperiments ? remoteExperiments.length : 0));
     }
 
+    protected handleTabNameUpdate(tabEditingOpen: string, index: number): void {
```

Comment:
All functions or events related to experiment name change should not refer to the tab.
`submitNewTab` -> `submitNewExperimentName`
`handleTabNameUpdate` -> `handleExperimentNameUpdate`
`fireTabChangedSignal` -> `fireExperimentChangedSignal`
`TABNAME_CHANGED` -> `EXPERIMENT_CHANGED`
`onTabNameChange` -> `onExperimentNameChange`
`doHandleTabeNameChange` -> `doHandleExperimentNameChange`
`tabEditingOpen` -> `newExperimentName`